pandas/tests/strings/test_strings.py

```diff
@@ -799,3 +799,19 @@ def test_str_accessor_in_apply_func():
     expected = Series(["A/D", "B/E", "C/F"])
     result = df.apply(lambda f: "/".join(f.str.upper()), axis=1)
     tm.assert_series_equal(result, expected)
+
+
+def test_zfill():
+    # https://github.com/pandas-dev/pandas/issues/20868
+    value = Series(["-2", "+5"])
+    wid = "a"
+    msg = f"width must be of integer type, not {type(wid).__name__}"
+    with pytest.raises(TypeError, match=msg):
```

Comment:
1. Could you make this TypeError test a separate test function?
2. Could you also add another test function test for data like `["-cat", "-1", "+dog"]`

Reply:
Sure, so there are going to be 3 tests in total? One test for general strings, one test error rasing, the last one test for strings with signs?

Reply:
Correct 3 total tests